cloudify_agent/operations.py

```diff
@@ -195,17 +195,14 @@ def _get_broker_url(ctx, agent):
 
 
 def create_new_agent_dict(old_agent):
-    new_agent = copy.deepcopy(old_agent)
-    fields_to_delete = ['name', 'queue', 'workdir', 'agent_dir', 'envdir',
-                        'manager_ip', 'package_url', 'source_url'
-                        'manager_port', 'broker_url', 'broker_port',
-                        'broker_ip', 'key', 'version']
-    for field in fields_to_delete:
-        if field in new_agent:
-            del(new_agent[field])
+    new_agent = {}
     new_agent['name'] = utils.internal.generate_new_agent_name(
         old_agent['name'])
     new_agent['remote_execution'] = True
+    fields_to_copy = ['windows', 'ip', 'basedir', 'user']
```

Comment:
Not exactly.
reinstallation_attributes is working differently.
Your usual configuration function first adds data from runtime properties, then properties, then defaults and if everything else fails code of the function is expected to fill in the gaps.
reinstallation_attributes on the other hand gives priority to the function code, and uses runtime properties/properties as fallback.
Otherwise we would end up installing agent in directory where old agent was installed.

Quick analysis of the code suggests that it would be sufficient to copy `windows` field.